src/cli/test/parse-flags.spec.ts

```diff
@@ -132,6 +132,39 @@ describe('parseFlags', () => {
     expect(flags.config).toBe('/config-2.js');
   });
 
+  describe('boolean-string flag', () => {
+    it('parses a boolean-string flag as a boolean with no arg', () => {
+      const args = ['--headless'];
+      const flags = parseFlags(args);
+      expect(flags.headless).toBe(true);
+      expect(flags.knownArgs).toEqual(['--headless']);
+    });
+
+    it.each([['--noHeadless'], ['--no-headless']])(
+      'parses a boolean-string flag as a falsy boolean with "no" arg - \'%s\'',
+      (noVariant) => {
+        const args = [noVariant];
+        const flags = parseFlags(args);
+        expect(flags.headless).toBe(false);
+        expect(flags.knownArgs).toEqual([noVariant]);
+      }
+    );
+
+    it("parses a boolean-string flag as a string with 'new' arg", () => {
+      const args = ['--headless', 'new'];
+      const flags = parseFlags(args);
+      expect(flags.headless).toBe('new');
+      expect(flags.knownArgs).toEqual(['--headless', 'new']);
+    });
+
+    it("parses a boolean-string flag as a string with 'new' arg using equality", () => {
+      const args = ['--headless=new'];
+      const flags = parseFlags(args);
+      expect(flags.headless).toBe('new');
+      expect(flags.knownArgs).toEqual(['--headless', 'new']);
+    });
+  });
```

Comment:
I know we only have one flag in this array right now, but what would you think about factoring this test as a `describe.each` with `BOOLEAN_STRING_CLI_FLAGS`, like we do for the other lists of flags? see [here](https://github.com/ionic-team/stencil/blob/9690a8226424db4bf2da4540bfdf826980c916ac/src/cli/test/parse-flags.spec.ts#L302-L309) for instance. I know we only have the one flag right now, but it's just nice to have assurance that if we add another in the future we'll always have full coverage

Reply:
I considered doing this too, I'm up for it!